src/main/java/com/purbon/kafka/topology/TopologyValidator.java

```diff
@@ -100,8 +101,12 @@ public TopologyValidator(Configuration config) {
                           validationClass));
                 }
 
-                Constructor<?> constructor = clazz.getConstructor();
-                Object instance = constructor.newInstance();
+                String constructorParam = getValidationStringParam(validationClass);
+                Object instance =
+                    constructorParam == null
+                        ? clazz.getConstructor().newInstance()
+                        : clazz.getConstructor(String.class).newInstance(constructorParam);
```

Comment:
_idea_:
  
If I understand your proposal accurate, you're building out the regexp pattern out of the same config attribute where you extract the validation class.

I wonder, what about:

* at this point the application configuration object is already available. In this object, we know all things that can define the behaviour of the application.
* we could define a config variable, remember we don't need to pre-register them, it could be something the Validation gets by using _config.getProperty(...)_.

Like this we would not need to:

* Use a separator when defining the validation chain.
* Use as many configs within validations as need without having to extend the constructor.

Does this makes sense to you?

Reply:
If there is another way I'm very happy to implement it differently. Not sure though if I understand correctly what you mean...

> at this point the application configuration object is already available. In this object, we know all things that can define the behaviour of the application. 

We can configure the Topology Builder to use multiple Validations, each of them potentially needing its own configuration parameters. How can we supply dynamic configuration options for an unknown number of Validations?

> we could define a config variable, remember we don't need to pre-register them, it could be something the Validation gets by using config.getProperty(...).

Same problem? Since we potentially deal with multiple Validations and thus potentially multiple configuration options, how would this work?

Just trying to understand...

Reply:
Hi Leonardo, I try to explain.

The application (JulieOps) will get a config file pass by the user, this is the same file you pass the configuration of the validation and it's order. Each of these instances could get during the construction time, so when you call the _newInstance/newConstructor_ methods the _Configuration_ object who is mainly a Java wrapper of that file.

In your proposal you suggest a sintax like:

```java
topology.validations.0=topology.TopicNameRegexValidation#foo.bar.[a-zA-Z]+.[a-zA-Z]+
```

that would clearly work and solve the problem. However, I was wondering if it would be nicer to have something like

```java
topology.validations.0=topology.TopicNameRegexValidation
topology.validations.regexp.0=foo.bar.[a-zA-Z]+.[a-zA-Z]+
```

indicating the second like that the regexp pattern for the first validation would be the given one.

The concern or thoughts I am having is around the need for the separator and how we can validate that people don't forget to add it.

I think the second proposal would make:

* The code to create the validations more easy and flexible. We just need to create the validation and pass the configuration object.
* We could add and use as many configurations as required for each validation.

BTW, JulieOps use the https://github.com/lightbend/config library for this purpose, there is no specific need to register any config, you can just call config.getProperty or other methods to fetch it and use it. 

Does this makes sense?